crates/precompile/src/bls12_381_const.rs

```diff
@@ -61,3 +38,111 @@ pub static DISCOUNT_TABLE_G2_MSM: [u16; 128] = [
     548, 547, 546, 545, 545, 544, 543, 542, 541, 541, 540, 539, 538, 537, 537, 536, 535, 535, 534,
     533, 532, 532, 531, 530, 530, 529, 528, 528, 527, 526, 526, 525, 524, 524,
 ];
+
+// Constants related to the bls12-381 precompile inputs and outputs
+
+/// FP_LENGTH specifies the number of bytes needed to represent an
+/// Fp element. This is an element in the base field of BLS12-381.
+///
+/// Note: The base field is used to define G1 and G2 elements.
+pub const FP_LENGTH: usize = 48;
+/// PADDED_FP_LENGTH specifies the number of bytes that the EVM will use
+/// to represent an Fp element according to EIP-2537.
+///
+/// Note: We only need FP_LENGTH number of bytes to represent it,
+/// but we pad the byte representation to be 32 byte aligned as specified in EIP 2537.
+pub const PADDED_FP_LENGTH: usize = 64;
+
+/// G1_LENGTH specifies the number of bytes needed to represent a G1 element.
+///
+/// Note: A G1 element contains 2 Fp elements.
+pub const G1_LENGTH: usize = 2 * FP_LENGTH;
+/// PADDED_G1_LENGTH specifies the number of bytes that the EVM will use to represent
+/// a G1 element according to padding rules specified in EIP-2537.
+const PADDED_G1_LENGTH: usize = 2 * PADDED_FP_LENGTH;
+
+/// FP2_LENGTH specifies the number of bytes needed to represent a
+/// Fp^2 element.
+///
+/// Note: This is the quadratic extension of Fp, and by definition
+/// means we need 2 Fp elements.
+const FP2_LENGTH: usize = 2 * FP_LENGTH;
+/// PADDED_FP2_LENGTH specifies the number of bytes that the EVM will use to represent
+/// a Fp^2 element according to the padding rules specified in EIP-2537.
+pub const PADDED_FP2_LENGTH: usize = 2 * PADDED_FP_LENGTH;
+
+/// SCALAR_LENGTH specifies the number of bytes needed to represent an Fr element.
+/// This is an element in the scalar field of BLS12-381.
+///
+/// Note: Since it is already 32 byte aligned, there is no padded version of this constant.
+/// TODO: Maybe change all _LENGTH to _LENGTH_BYTES and then NBITS to _LENGTH_BITS
+pub const SCALAR_LENGTH: usize = 32;
+/// NBITS specifies the number of bits needed to represent an Fr element.
+/// This is an element in the scalar field of BLS12-381.
+/// TODO: We can likely remove this and just check what the min(scalar.length()) is before doing an MSM
```

Comment:
Remove TODO, this is an optimization that does not need to be done and is unrelated to documenting the constants

Reply:
hm probably oversight

Reply:
will merge this PR with that TODO